clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -579,6 +579,17 @@ static void instantiateSYCLIntelPipeIOAttr(
     S.addSYCLIntelPipeIOAttr(New, *Attr, Result.getAs<Expr>());
 }
 
+static void instantiateIntelReqdSubGroupSizeAttr(
+    Sema &S, const MultiLevelTemplateArgumentList &TemplateArgs,
+    const IntelReqdSubGroupSizeAttr *Attr, Decl *New) {
+  // The SubGroupSize expression is a constant expressions.
+  EnterExpressionEvaluationContext Unevaluated(
+      S, Sema::ExpressionEvaluationContext::ConstantEvaluated);
+  ExprResult Result = S.SubstExpr(Attr->getSubGroupSize(), TemplateArgs);
+  if (!Result.isInvalid())
+    S.addIntelReqdSubGroupSizeAttr(New, *Attr, Result.getAs<Expr>());
+}
```

Comment:
This comment shouldn't block this particular patch since is a general problem, but I want to highlight it.
If you take a look at the code above, you will feel deja vu, because there are a ton of same copy-pasted functions for other SYCL attributes (most of them for FPGA). I think we have to get rid of this copy-paste and unify same code across different attribute handling.